### PR TITLE
Simple post creation/validation, display all posts.

### PR DESCRIPTION
**Pull request by izzipizzi**

### CODE REVIEW

File: src/app/components/post-form/post-form.component.scss [hunk 1]
```diff
@@ -0,0 +1,49 @@
+.add-post-form-wrapper {
```
Comment (MMaksymivUa) [inline src/app/components/post-form/post-form.component.scss:1]: Better to use the BEM approach

File: src/app/components/post-form/post-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+@Component({
+  selector: 'app-post-form',
+  templateUrl: './post-form.component.html',
+  styleUrls: ['./post-form.component.scss']
+})
+export class PostFormComponent implements OnInit {
+  newPostForm: FormGroup;
+  maxTitleLength: number = 60;
+  minTextLength: number = 3;
+  maxTextLength: number = 7000;
```
Comment (MMaksymivUa) [inline src/app/components/post-form/post-form.component.ts:15]: If you want to store it as a variable - add `private readonly` modificator

File: src/app/posts/models/Post.model.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  }
+}
+
+export class CreatePostPartialData {
+  constructor(
+    public title: string,
+    public text: string,
+  ) {
+  }
+}
```
Comment (MMaksymivUa) [inline src/app/posts/models/Post.model.ts:20]: You can create interface Post
```
export class CreatePostPartialModel implements Partial<Post>{}
or use Pick
export class PostModel implements Post{ }
```

CreatePost and Post should be linked(inheritance, composition)

File: src/app/posts/post/post.component.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      </div>
+      <div class="likes-container flex-row-center" [ngClass]="postLikes">
+        <img src="assets/svg/heart.svg" alt="Heart">
+        <span class="likes-indicator">{{post?.likes}}</span>
+      </div>
+    </div>
+    <div class="post-info-container">
+      <span class="post-title">{{post?.title}}</span>
+      <span class="post-text" [ngClass]="{'opened-text': isOpened}">{{post?.text}}</span>
+      <span class="post-btn" (click)="showMore()" *ngIf="!isOpened">Read more...</span>
```
Comment (MMaksymivUa) [inline src/app/posts/post/post.component.html:19]: change the text only in depending of condition
```
{{isOpened ? 'Show less' : 'Read more'}}
```

File: src/app/posts/post/post.component.scss [hunk 5]
```diff
@@ -0,0 +1,66 @@
+.post-wrapper {
```
Comment (MMaksymivUa) [inline src/app/posts/post/post.component.scss:1]: https://en.bem.info/

File: src/app/posts/services/posts.service.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import {Injectable} from '@angular/core';
+import {CreatePostPartialData, Post} from "../models/Post.model";
+import {Author} from "../models/Author.model";
+import {BehaviorSubject, Observable, of} from "rxjs";
```
Comment (MMaksymivUa) [inline src/app/posts/services/posts.service.ts:4]: Use prettier for the project

File: tsconfig.json [hunk 7]
```diff
@@ -27,6 +27,7 @@
     "enableI18nLegacyMessageIdFormat": false,
     "strictInjectionParameters": true,
     "strictInputAccessModifiers": true,
-    "strictTemplates": true
+    "strictTemplates": true,
+    "enableIvy": false
```
Comment (MMaksymivUa) [inline tsconfig.json:31]: Why did you do that?

File: src/app/posts/post/post.component.html (hunk 4 above)
Comment (MMaksymivUa) [reply src/app/posts/post/post.component.html:19]: and simple add actionHandler
isOpened = !isOpened

File: src/app/components/post-form/post-form.component.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  get textError(): string {
+    const formControl = this.newPostForm.get('text');
+    if (formControl?.hasError('required')) {
+      return 'Please enter some text.'
+    } else if (formControl?.hasError('minlength')) {
+      return `The text must be longer than ${this.minTextLength} characters.`
+    } else if (formControl?.hasError('maxlength')) {
+      return `The text must be no longer than ${this.maxTextLength} characters.`
+    }
+    return ''
```
Comment (MMaksymivUa) [inline src/app/components/post-form/post-form.component.ts:63]: btw: You can create a Map<ErrorName, ErrorDescription>

File: tsconfig.json (hunk 7 above)
Comment (izzipizzi) [reply tsconfig.json:31]: Angular show me error, when i use pipes. This is fix.

**Pull request merged**